system-addon/content-src/components/StartupOverlay/StartupOverlay.jsx

```diff
@@ -0,0 +1,87 @@
+import {actionCreators as ac} from "common/Actions.jsm";
+import {connect} from "react-redux";
+import {injectIntl} from "react-intl";
+import React from "react";
+
+export class _StartupOverlay extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.locations = {};
+    this.onInputChange = this.onInputChange.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
+    this.clickSkip = this.clickSkip.bind(this);
+    this.initScene = this.initScene.bind(this);
+
+    this.state = {emailInput: ""};
+  }
+
+  initScene(element) {
+    document.body.dataset.welcome = "true";
+    this.scene = element;
+    // Timout to allow the scene to render once before attaching the attribute
+    // to trigger the animation.
+    setTimeout(() => {
+      this.scene.dataset.content = "true";
+    }, 0);
+  }
+
+  removeOverlay(e) {
+    let wrapper = document.querySelector(".overlay-wrapper");
+    wrapper.dataset.signIn = "true";
+    setTimeout(() => {
+      // Allow scrolling and remove overlay.
+      delete document.body.dataset.welcome;
+    }, 400);
+  }
+
+  onInputChange(e) {
+    this.setState({emailInput: e.target.value});
+  }
+
+  onSubmit() {
+    this.props.dispatch(ac.UserEvent({event: "SUBMIT_EMAIL"}));
+    window.addEventListener("focus", this.removeOverlay);
+  }
+
+  clickSkip() {
+    this.props.dispatch(ac.UserEvent({event: "CLICK_SKIP"}));
+    this.removeOverlay();
+  }
+
+  render() {
+    return (
+      <div className="overlay-wrapper">
+        <canvas className="canvas">
+          HTML5 is not supported.
+        </canvas>
+        <div className="firstrun-scene" ref={this.initScene}>
+          <div className="fxaccounts-container">
+            <div className="firstrun-left-divider">
+              <h1 className="firstrun-title">Take Firefox with You</h1>
+              <p className="firstrun-content">Get your bookmarks, history, passwords and other settings on all your devices.</p>
+              <a className="firstrun-link" href="https://www.mozilla.org/firefox/features/sync/" target="_blank" rel="noopener noreferrer">Learn more about Firefox Accounts</a>
+            </div>
+            <div className="firstrun-sign-in">
+              <p className="form-header">Enter your email <span>to continue to Firefox Sync</span></p>
+              <form method="get" action="https://accounts.firefox.com" target="_blank" rel="noopener noreferrer" onSubmit={this.onSubmit}>
+                <input name="service" type="hidden" value="sync" />
```

Comment:
All these inputs are hidden, any reason we can't remove them?

Reply:
Good question, I'll have to ask @vladikoff about that.

Reply:
Heya, remove if you want, just make sure you pass the params to fxa in some other way so users can login to Sync